pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -313,9 +300,13 @@ func (h *AutoscalersController) syncAutoscalers(key interface{}) error {
 		}
 		new := h.hpaProc.ProcessHPAs(hpa)
 		h.toStore.m.Lock()
-		h.toStore.data = append(h.toStore.data, new...)
-		log.Tracef("Local batch cache of HPA is %v", h.toStore.data)
+		for e, i := range new {
```

Comment:
```suggestion
		for metric, value := range h.hpaProc.ProcessHPAs(hpa) {
```

Would improve readability IMHO